### ISSUE TITLE
Catboost multiclass inference on an ESP32

### ISSUE THREAD

**Issue by AlverGant** [catboost/catboost#2506]
#### Multiclass Inference on ESP32 with C++: Probabilities Appear Inverted

Hi CatBoost team!

I'm facing an issue while trying to perform multiclass (2 classes) inference with CatBoost on an ESP32 using C++. I'm applying a softmax function to the model's output on the ESP32 to get a vector of probabilities. However, the probabilities appear to be inverted, and I'd like to confirm whether this is a known issue or if it's not supported at all. I'm hoping to get some guidance on how to properly implement this.

For context, I've previously had success running a RandomForest classifier on the ESP32, but I find CatBoost to be far superior and would love to use it on this platform.

#### Code on ESP32

Here's the C++ code snippet I'm using on the ESP32:

```cpp
#include "catboost_model.h"

std::vector<double> softmax(const std::vector<double>& logits) {
  std::vector<double> exp_logits(logits.size());
  for (size_t i = 0; i < logits.size(); ++i) {
    exp_logits[i] = std::exp(logits[i]);
  }
  double sum_exp = std::accumulate(exp_logits.begin(), exp_logits.end(), 0.0);
  std::vector<double> softmax_values(logits.size());
  for (size_t i = 0; i < logits.size(); ++i) {
    softmax_values[i] = exp_logits[i] / sum_exp;
  }
  return softmax_values;
}

void makePrediction(float t, float h, float p, float c) {
  std::vector<float> floatFeatures = {
    t, p, h, c
  };
  double singleLogit = ApplyCatboostModel(floatFeatures);
  std::vector<double> logits = {singleLogit, 0.0};
  std::vector<double> local_probabilities = softmax(logits);
  // ... (printing the results)
}
```

#### Environment Details

- **CatBoost Version**: 1.2.2
- **Operating System**: Linux Mint 21 / ESP32
- **CPU**: Intel(R) Xeon(R) CPU E3-1270 v3 @ 3.50GHz / ESP32
- **GPU**: NVIDIA GeForce RTX 3060

Would appreciate any help or guidance on this matter!

Thank you!

**Comment by andrey-khropov** [comment]
If you train a binary classification model then the raw model output returned by `ApplyCatboostModel` will be the logit for the positive class, so if you want to get probabilities in the order (negative class, positive class) then you just have to change the line initializing logits in your code to:

```c++
std::vector<double> logits = {0.0, singleLogit};
```

**Comment by AlverGant** [comment]
Thank you @andrey-khropov it is working now!

**Comment by andrey-khropov** [comment]
> Thank you @andrey-khropov it is working now!

So, can this issue be closed then?

**Comment by AlverGant** [comment]
Yes, we can close the issue
